### ISSUE TITLE
Creating a unique index on partition changes distribution key

### ISSUE THREAD

**Issue by hlinnaka** [greenplum-db/gpdb#10376]
### Greenplum version or build
5X_STABLE-master

### Step to reproduce the behavior

```
postgres=# create table foopart (i int4, j int4) distributed by (i) partition by range (i) (start (1) end (3) every (1));
psql: NOTICE:  CREATE TABLE will create partition "foopart_1_prt_1" for table "foopart"
psql: NOTICE:  CREATE TABLE will create partition "foopart_1_prt_2" for table "foopart"
CREATE TABLE
postgres=# create unique index on foopart_1_prt_1 (j);
psql: NOTICE:  updating distribution policy to match new UNIQUE index
CREATE INDEX
```
Changing the distribution key of the partition is not OK, because a partition hierarchy is required to have the same distribution key in all partitions. We don't allow that with ALTER TABLE EXCHANGE PARTITION, for example.

I also noticed that pg_dump doesn't dump the index on the partition at all.

**Comment by pivotal-ning-yu** [comment]
Nice catch, this is very bad behavior.  I'll work on this, thanks.

**Comment by gfphoenix78** [comment]
This was an expected behavior in the test partindex_test. I removed it in the PR https://github.com/greenplum-db/gpdb/pull/10510

**Comment by hlinnaka** [comment]
This was fixed in commit 84d2a23fbd0f81c6c4b372701a27a6dfbada8613 (PR #10510), which disabled implicitly changing distribution keys on CREATE UNIQUE INDEX altogether.
